diff --git a/hbjs.cc b/hbjs.cc
index 946ab2d..849a43b 100644
--- a/hbjs.cc
+++ b/hbjs.cc
@@ -190,6 +190,11 @@ static hb_bool_t do_trace (hb_buffer_t *buffer,
     HB_BUFFER_SERIALIZE_FORMAT_JSON,
     HB_BUFFER_SERIALIZE_FLAG_NO_GLYPH_NAMES);
   user_data->consumed += consumed;
+  if (hb_buffer_get_content_type(buffer) == HB_BUFFER_CONTENT_TYPE_GLYPHS) {
+    _user_data_printf (user_data, ", \"glyphs\": true");
+  } else {
+    _user_data_printf (user_data, ", \"glyphs\": false");
+  }
   _user_data_printf (user_data, "},\n");

   return 1;